### PR TITLE
Command line script and renaming Brackets.sh -> brackets.sh

### PR DESCRIPTION
**Pull request by nethip**
This change contains the following changes 
* Renames Brackets.sh to brackets.sh 
* Made some changes to script that fixes a unit testing bug
* Update the gyp script to copy brackets.sh instead of Brackets.sh

### CODE REVIEW

File: appshell/appshell_extensions_mac.mm [hunk 1]
```diff
@@ -1226,6 +1226,25 @@ int32 InstallCommandLineTools()
...
+        
+        sourcePath = [[NSBundle mainBundle] pathForResource: @"brackets" ofType: @"sh"];
+        if(!sourcePath) {
+            // If we have not found this in the bundle try the hard path.
+            NSString* bundlePath = [[NSBundle mainBundle] bundlePath];
+            sourcePath = [bundlePath stringByAppendingString:@"/Contents/Resources/brackets.sh"];
+        }
+        
+        if(!sourcePath)
+            return ERR_FILE_NOT_FOUND;
```
Comment (peterflynn) [inline appshell/appshell_extensions_mac.mm]: Can this ever happen?  In other words, can `stringByAppendingString` ever return null?

File: appshell/mac/brackets.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+TARGET_FILE=$0
+BRACKETS_APP_NAME="Brackets.app"
+
+TARGET_PATH="$(readlink "$TARGET_FILE")"
```
Comment (peterflynn) [inline appshell/mac/brackets.sh]: So if I'm reading this correctly, `$0` will be the symlink (`/usr/local/bin/brackets`), and `TARGET_PATH` will be the actual location of this .sh file, inside the Brackets.app bundle?  Could we add a comment here explaining that?

File: appshell/mac/brackets.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    FINAL_APP_PATH="";
+fi
+
+# Give preference to launching the script's enclosing app
+if [ -x "$FINAL_APP_PATH" ]; then
+    open -a "$FINAL_APP_PATH" "$@"
+elif [ -x "/Applications/$BRACKETS_APP_NAME" ]; then
+    open -a /Applications/$BRACKETS_APP_NAME "$@"
+elif [ -x "$HOME/Applications/$BRACKETS_APP_NAME" ]; then
+    open -a $HOME/Applications/$BRACKETS_APP_NAME "$@"
```
Comment (peterflynn) [inline appshell/mac/brackets.sh]: Seems like this should take precedence over the global /Applications

**Comment by peterflynn** [comment]
@nethip A couple small items here, but after that it should be good to merge.

File: appshell/appshell_extensions_mac.mm (hunk 1 above)
Comment (nethip) [inline appshell/appshell_extensions_mac.mm]: The documentation says `stringByAppendingString` could return `null in failure scenarios. So a check here would be better for sure.

File: appshell/mac/brackets.sh (hunk 2 above)
Comment (nethip) [inline appshell/mac/brackets.sh]: I will put more comments to this file.

File: appshell/mac/brackets.sh (hunk 3 above)
Comment (nethip) [inline appshell/mac/brackets.sh]: Will change this.

Comment (nethip) [inline appshell/mac/brackets.sh]: Chances of someone running into the first condition failure are pretty less, unless someone creates a symlink to brackets.sh outside the .app(that would be without using our `install` procedure). For e.g.: After the symlink is created, if the app, from which the symlink was created, is moved or renamed, typing `brackets` would give a `-bash /usr/local/bin/brackets: no file or directory` error, as the target cannot be reached.

Comment (peterflynn) [inline appshell/mac/brackets.sh]: That's a good point... and Launch Services is probably smart enough to get the user-Applications vs. global-Applications search order right on its own too, so probably all we really need are the first case and the last case.  But the code as-is now seems fine too.

**Comment by peterflynn** [comment]
Looks good now -- merging

**Pull request merged**